src/Symfony/Bridge/Twig/AppVariable.php

```diff
@@ -174,6 +174,13 @@ public function getFlashes($types = null)
             return $session->getFlashBag()->get($types);
         }
 
-        return array_intersect_key($session->getFlashBag()->all(), array_flip($types));
+        $result = array();
+        foreach ($types as $type) {
+            if ($value = $session->getFlashBag()->get($type)) {
```

Comment:
Don't we want to return an empty array for types for which `$value` evaluates to `false` here?